CMSgov/design-system
# [WNMGDS-1591] prop table

## Summary

* Added page width to page content to align with current doc site max width
* Installed gatsby-react-docgen plugin to help query React components and get prop data
  * This did throw some warnings in the design-system repo about "Un-closed Components" which is why I added tick marks to some of the comments
* I made a MockComponentPage to display the PropTable component. Some form of this will be converted to a Component layout page, but in a future ticket
* I made a PropTable component to display prop data.

### How Data Flow Works

* Component Page loads the prop data with the `loadComponentPropsQuery`. Right now `"Alert"` is the hard-coded component name, but the intention is to update this to a `$componentName` variable
  * In order to have a variable in a query, the query must happen in a "page" and not a "component" which is why the PropTable doesn't do its own query for data. Read more about querying in [pages](https://www.gatsbyjs.com/docs/how-to/querying-data/page-query/) & [components](https://www.gatsbyjs.com/docs/how-to/querying-data/use-static-query/)


## How to test

`yarn install` & `yarn start:gatsby`. Navigate to this URL by changing URL (not in nav bar as it is a test page): http://localhost:8000/test/MockComponentPage/ . See prop table for an Alert component

## Things I'm pondering for a future ticket
* How does the prop table work with MDX?
  * Can't import the PropTable into an .mdx file because it needs prop data from somewhere and the component can't load that data itself because component queries can't have variables in them
  * Trying to figure out if there's something that can be done in ContentRenderer to help replace a token and plop the PropTable in
  * Otherwise, it would be great if the PropTable could exist at the end of the markdown content as the last thing on the page.
